projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -320,6 +327,33 @@ def pre_ingest_process(self, payload: List[dict]) -> List[dict]:
         :param metadata: Optional[IngestionMetadata] dict object containing
         metadata produced by the pre-ingest plugin, and possibly used by
         other pre-ingest, ingest, or post-ingest plugins.
+        :param destination_table: Optional[string]
+            (Optional) The name of the table, where the data should be
+            ingested into. This argument is optional, and needs to be
+            considered only when the name of the destination table is not
+            included in the payload itself.
+            NOTE: A read-only parameter. It is not expected to be modified and
```

Comment:
is any parameter mutable, for example `payload`?

Reply:
Yes, payload and metadata are mutable. The notes are only for read-only parameters, that are not expected to be modified.

Reply:
Aren't all parameters in fact immutable?
Isn't the modified payload and metadata rather returned??

Reply:
Correct, from design point of view they are immutable.